torchvision/tv_tensors/_mask.py

```diff
@@ -37,3 +37,34 @@ def __new__(
 
         tensor = cls._to_tensor(data, dtype=dtype, device=device, requires_grad=requires_grad)
         return tensor.as_subclass(cls)
+
+
+class SegmentationMask(Mask):
+    def __new__(
+        cls,
+        data: Any,
+        *,
+        dtype: Optional[torch.dtype] = None,
+        device: Optional[Union[torch.device, str, int]] = None,
+        requires_grad: Optional[bool] = None,
+    ) -> Mask:
+        if isinstance(data, PIL.Image.Image):
+            from torchvision.transforms.v2 import functional as F
+
+            data = F.pil_to_tensor(data)
+
+        tensor = cls._to_tensor(data, dtype=dtype, device=device, requires_grad=requires_grad)
+        return tensor.as_subclass(cls)
+
+
+class DetectionMasks(Mask):
+    def __new__(
+        cls,
+        data: Any,
+        *,
+        dtype: Optional[torch.dtype] = None,
+        device: Optional[Union[torch.device, str, int]] = None,
+        requires_grad: Optional[bool] = None,
+    ) -> Mask:
+        tensor = cls._to_tensor(data, dtype=dtype, device=device, requires_grad=requires_grad)
```

Comment:
We should probably have similar shape checks here as

https://github.com/pytorch/vision/blob/a06df0d9229e49bd859e2ff0355a48b3bddd1e10/torchvision/tv_tensors/_bounding_box.py#L53-L57

Reply:
The rationale to prevent instantiation from PIL images sounds OK to me, but we might as well raise an informative error message when passed a PIL image then